File: LeetCode_C++/0045.跳跃游戏-ii.cpp
/*
 * @lc app=leetcode.cn id=45 lang=cpp
 *
 * [45] 跳跃游戏 II
 */

// @lc code=start
class Solution {
public:
    int jump(vector<int>& nums) {
        int n = nums.size();
        vector<int> f(n, 0);
        for(int i = 1, j = 0; i < n; i++) {
            while(j + nums[j] < i) j++;
            f[i] = f[j] + 1;
        }
        return f[n - 1];
    }
};
// @lc code=end

File: LeetCode_Java/0045.跳跃游戏-ii.java
/*
 * @lc app=leetcode.cn id=45 lang=java
 *
 * [45] 跳跃游戏 II
 */

// @lc code=start
class Solution {
    public int jump(int[] nums) {
        int n = nums.length;
        int[] f = new int[n];
        for (int i = 1, last = 0; i < n; i++) {
            while (last + nums[last] < i) last++;
            f[i] = f[last] + 1;
        }
        return f[n - 1];
    }
}
// @lc code=end

